astropy/coordinates/representation.py

```diff
@@ -483,7 +554,141 @@ def represent_as(self, other_class):
+
+            else:
+                new_rep = other_class.from_cartesian(self.to_cartesian())
+
+            return new_rep
+
+    def with_differentials(self, differentials):
+        """
+        Create a new representation with the same positions as this
+        representation, but with new differentials.
+
+        Parameters
+        ----------
+        differentials : Sequence of `~astropy.coordinates.BaseDifferential`
+            The differentials for the new representation to have.
+
+        Returns
+        -------
+        newrepr
+            A copy of this representation, but with the ``differentials`` as
+            its differentials.
+        """
+        # Implemented this way for (1) performance (skips going through the
+        # initializer), and (2) will preserve any additional attributes a user
+        # may have set
+
+        new_differentials = self._validate_differentials(differentials)
+
+        # save the differentials, then strip them so that the copy doesn't do
+        # an unneeded copy, then re-attach them to the right objects
+        olddiffs = self._differentials
+        try:
+            self._differentials = None
+            newrepr = deepcopy(self)
+            newrepr._differentials = new_differentials
+            return newrepr
+        finally:
+            self._differentials = olddiffs
+
+    def without_differentials(self):
+        """
+        Create a new representation with the same positions as this
+        representation, but without any differentials.
+
+        Returns
+        -------
+        newrepr
+            A copy of this representation, but without the ``differentials``.
+        """
+
+        # Again, implemented this way for (1) performance (skips going through
+        # the initializer), and (2) will preserve any additional attributes a
+        # user may have set
+
+        # save the differentials, then strip them so that the copy doesn't do
+        # an unneeded copy, then re-attach them to the right objects
+        olddiffs = self._differentials
+        try:
+            self._differentials = None
+            newrepr = deepcopy(self)
+            newrepr._differentials = ()
+            return newrepr
+        finally:
+            self._differentials = olddiffs
+
+    def to_cartesian(self, diffstocart=False):
+        """
+        Convert the representation to its carteisan form.
+
+        Note that subclasses should *not* override this - rather they should
+        override the ``_to_cartesian_helper()`` method.
+
+        Parameters
+        ----------
+        diffstocart : bool
+          If True, any differentials this representation has are *also*
+          converted to their cartesian form.
+
+        Returns
+        -------
+        cartrepr : CartesianRepresentation
+          The representation in Cartesian form.
+        """
+        repr = self._to_cartesian_helper()
+        if diffstocart:
+            repr._differentials = tuple([
```

Comment:
Tuple comprehension doesn't work directly, it only seems to work by doing, e.g.,
```python
tuple(x for x in range(10))
```
because `x for x in range(10)` produces a generator, which then gets turned into a tuple. So it's actually marginally faster to pass in a list comprehension to `tuple()` because the conversion is faster. But we're talking about a 20% speedup at the microsecond level, so I don't really care either way if you think it looks cleaner without the bracket...